index.js

```diff
@@ -1,43 +1,62 @@
 /* eslint-env node */
 'use strict';
+const Funnel = require('broccoli-funnel');
+const getDebugMacros = require('./src/debug-macros').debugMacros;
 
-function debugMacros(features) {
-  let plugins = [
-    [
-      require.resolve('babel-plugin-debug-macros'),
-      {
-        flags: [
-          {
-            source: 'ember-m3/feature-flags',
-            flags: features,
-          },
-        ],
-      },
-      '@ember-data/canary-features-stripping',
-    ],
-  ];
-
-  return plugins;
+/*
+ shouldIncludeChildAddon runs during
+ addon init so does not have access to
+ the host application instance.
+*/
+function hasDataPackage(addon) {
+  let addons = addon.project.addonPackages;
+  return addons['ember-data'] !== undefined;
 }
 
 module.exports = {
   name: 'ember-m3',
 
-  init() {
-    this._super.init.apply(this, arguments);
+  included() {
+    this._super.included.call(this, ...arguments);
+    this.configureBabelOptions();
+  },
+
+  shouldIncludeChildAddon(addon) {
+    if (!hasDataPackage(this)) {
+      return true;
+    }
+    if (addon.name.startsWith('@ember-data')) {
+      /*
+      console.log(
+        `⚠️  ember-m3 is excluding ${addon.name} version ${addon.pkg.version} from the build`
+      );
+      */
+      return false;
+    }
+    return true;
+  },
+
+  treeForAddon(tree) {
+    const isProd = process.env.EMBER_ENV === 'production';
+    const exclude = [];
 
-    let features;
-    try {
-      features = this.project.require('@ember-data/private-build-infra/src/features')();
-    } catch (e) {
-      features = { CUSTOM_MODEL_CLASS: false };
+    if (isProd) {
+      exclude.push('adapters');
     }
+    tree = new Funnel(tree, {
```

Comment:
This creates an un-needed funnel if we are in the else clause of the `isProd` guard.
A potential alternative would be:

```js
if (isProd) {
  tree = new Funnel(tree, { exclude, })
}
```